Docs: Add `Potential issues with Jest testing` React docs section (#17032)
ckeditor/ckeditor5: docs/getting-started/integrations/vuejs-v3.md

For more information, refer to the {@link getting-started/setup/ui-language Setting the UI language} guide.

> ### Potential issues with Jest testing

Reviewer: ```suggestion
## Known issues
### Potential issues with Jest testing
```

> Jest is the default test runner used by many Vue apps. Unfortunately, Jest does not use a real browser. Instead, it runs tests in Node.js with the use of JSDOM. JSDOM is not a complete DOM implementation and while it is apparently sufficient for standard apps, it is not able to polyfill all the DOM APIs that CKEditor&nbsp;5 requires.

Reviewer: If I'm not wrong Jest was the default but now it's vitest.
```suggestion
You can use Jest as a test runner in Vue apps. Unfortunately, Jest does not use a real browser. Instead, it runs tests in Node.js that uses JSDOM. JSDOM is not a complete DOM implementation, and while it is sufficient for standard apps, it cannot polyfill all the DOM APIs that CKEditor&nbsp;5 requires.
```

If this is not possible, you can use the following mocks to make the tests pass:

```javascript
beforeAll( () => {
	window.scrollTo = jest.fn();

	window.ResizeObserver = class ResizeObserver {
		observe() {}
		unobserve() {}
		disconnect() {}
	};

	for (const key of ['InputEvent', 'KeyboardEvent']) {
		window[key].prototype.getTargetRanges = () => {
			const range = new StaticRange({
				startContainer: document.body.querySelector('.ck-editor__editable p')!,
				startOffset: 0,
				endContainer: document.body.querySelector('.ck-editor__editable p')!,
				endOffset: 0,
			});

			return [range];
		};
	}

	Range.prototype.getClientRects = () => ({
		item: () => null,
		length: 0,
		[Symbol.iterator]: function* () {},
	});
} );
```

> These mocks should be placed before the tests that use CKEditor&nbsp;5. They are not perfect and may not cover all the cases, but they should be sufficient for basic initialization and rendering editor. Keep in mind that they are not a replacement for proper browser testing.

Reviewer: ```suggestion
These mocks should be placed before the tests that use CKEditor&nbsp;5. They are imperfect and may not cover all the cases, but they should be sufficient for basic initialization and rendering editor. Remember that they are not a replacement for proper browser testing.
```
